src/Shelly.hs

```diff
@@ -1056,11 +1056,15 @@ instance Exception RunFailed
 
 show_command :: FilePath -> [Text] -> Text
 show_command exe args =
-    T.intercalate " " $ map quote (toTextIgnore exe : args)
-  where
-    quote t | T.any (== '\'') t = t
-    quote t | T.any isSpace t = surround '\'' t
-    quote t | otherwise = t
+  let specialsInQuotes = Set.fromList $ ['\\', '"', '`', '$']
+      escape char | char `Set.member` specialsInQuotes = T.pack ['\\', char]
+      escape char = T.singleton char
+      quote arg = surround '"' $ T.concatMap escape arg
+      specials = Set.fromList "\\'\"`$&|;(){}<>"
```

Comment:
Lift this definition out to the toplevel, or can we be sure that GHC does this for us?